app/models/concerns/variant_stock.rb

```diff
@@ -130,6 +130,23 @@ def in_stock?(quantity = 1)
     can_supply?(quantity)
   end
 
+  # Moving Spree::StockLocation.fill_status to the variant enables us to override this behaviour for variant overrides
+  # We can have this responsibility here in the variant because there is only one stock item per variant
+  #
+  # Here we depend only on variant.total_on_hand and variant.on_demand.
+  #   This way, variant_overrides only need to override variant.total_on_hand and variant.on_demand.
+  def fill_status(quantity)
+    if count_on_hand >= quantity
+      on_hand = quantity
+      backordered = 0
+    else
+      on_hand = total_on_hand < 0 ? 0 : total_on_hand
```

Comment:
I find this one easier:
```suggestion
      on_hand = [0, total_on_hand].max
```

Reply:
spree code is actually:
```
          on_hand = item.count_on_hand
          on_hand = 0 if on_hand < 0
```

I'll keep your suggestion.